client/scala/src/main/scala/ai/verta/repository/Commit.scala

```diff
@@ -304,6 +304,32 @@ class Commit(
     )
   }
 
+  /** Reverts other.
+   *  This method creates and returns a new Commit in ModelDB, and assigns a new ID to this object
+   *  Currently reverting a merge commit is not supported. Unexpected behavior might occur.
+   *  @param other Base for the revert. If not provided, this commit will be reverted
+   *  @param message Description of the revert. If not provided, a default message will be used
+   *  @return The new commit, with the changes in other reverted, if suceeds. Failure if this commit or other has not yet been saved, or if they do not belong to the same Repository.
+   */
+  def revert(other: Commit = this, message: Option[String] = None)(implicit ec: ExecutionContext) = {
+    if (!saved)
+      Failure(new IllegalCommitSavedStateException("This commit must be saved"))
+    else if (!other.saved)
+      Failure(new IllegalCommitSavedStateException("Other commit must be saved"))
+    else if (other.repo.id != repo.id)
+      Failure(new IllegalArgumentException("Two commits must belong to the same repository"))
```

Comment:
Seems like we repeat this check in a few places. Can we extract this into its own lib?

Reply:
Hmm what do you have in mind? I can write a helper function that (optionally) take the other commit, check if either is unsaved (in which case a Failure is return). It can either just use a default error message (i.e "this commit is not saved", "other commit is not saved"), or require error messages to be passed to it. I'm not sure if it's worth the overhead though?

Reply:
I just make the changes in the latest commit. See if it looks good.